ui/src/app/google-analytics-events.service.ts

```diff
@@ -0,0 +1,19 @@
+import {Injectable} from "@angular/core";
+declare let gtag: Function;
+ 
+@Injectable()
+export class GoogleAnalyticsEventsService {
+ 
+  public emitEvent(eventName: string,
+		   eventCategory: string,
+                   eventAction: string,
+                   eventLabel: string = null,
+                   eventValue: number = null) {
+    gtag('event', eventName, {
+      eventCategory: eventCategory,
+      eventLabel: eventLabel,
+      eventAction: eventAction,
+      eventValue: eventValue
+    });
+  }
+}
```

Comment:
NIT: end the files with a newline please.

Reply:
Done.